client/jetpack-connect/authorize.js

```diff
@@ -888,7 +889,14 @@ export class JetpackAuthorize extends Component {
 		const productSlug = searchParams.get( 'productSlug' );
 		const siteSlug = searchParams.get( 'fromSiteSlug' );
 		const product = getProductFromSlug( productSlug );
-		const productName = getJetpackProductDisplayName( product );
+		let productName = '';
+
+		if ( product?.product_name ) {
+			productName = getJetpackProductDisplayName( product );
+		} else {
+			productName = getPlan( productSlug )?.getTitle();
+		}
+
```

Comment:
I'd probably shorten this into a ternary 
```suggestion
        const productName = product?.product_name ? getJetpackProductDisplayName( product ) : getPlan( productSlug )?.getTitle();
```
but that is a personal preference